### DISCUSSION TITLE
Patching poweshell on every machine and server

### DISCUSSION THREAD

**Discussion by NorthantsPete** [PowerShell/PowerShell#16282] [General]
https://msrc.microsoft.com/update-guide/vulnerability/CVE-2021-41355

The above security risk needs patching - the Q&A just describe updating one machine by downloading a new setup file but surely, PowerShell is installed on every server and PC in the world

Do Microsoft intend to patch this from windows update?
If not, how do they plan to assist rolling out to every known machine

Why hasnt it had more focus and a KB/hotfix via normal channels?

**Comment by iSazonov** [comment]
Only PowerShell 7.1 is affected, not Windows PowerShell. PowerShell 7.1 is not a Windows component and it doesn't present on every system. If an administrator installed PowerShell 7.1 he/she should use the same method to update the PowerShell 7.1.

Also see https://github.com/PowerShell/PowerShell/discussions/15510 - Windows Update works for PowerShell 7.1.
